client/containers/Users/Dialogs/RemoveMultiFactorDialog.jsx

```diff
@@ -47,6 +55,21 @@ export default connectContainer(class extends Component {
     const message = getDialogMessage(messageFormat, 'username',
       getName(user, userFields, languageDictionary));
 
+    const fields = _.cloneDeep(userFields) || [];
+    const providers = user && user.multifactor ? user.multifactor : [];
+    useMfaField(true, fields, providers);
+
+    const allowedFields = [ 'user_id', 'multifactor' ];
+    const filteredFields = _.filter(fields,
+      field => _.includes(allowedFields, field.property));
+
+    const UserFieldsFormInstance = UserFieldsForm('remove-mfa', this.onSubmit.bind(this));
+
+    const initialValues = mapValues(user, allowedFields, filteredFields, 'edit', languageDictionary);
+    if (initialValues.multifactor) {
+      initialValues.multifactor = JSON.parse(initialValues.multifactor)[0];
```

Comment:
any chance that this `JSON.parse` would fail?

Reply:
The check `if (initialValues.multifactor)` ensures that the `multifactor` value exists which comes directly from API2. The only way I can see this breaking is if API2 contract for fetching a user is broken.